tools/ci-ts/test-helpers/chainlinkClient.test.ts

```diff
@@ -0,0 +1,150 @@
+let containerMockState = {
+  Paused: false,
+}
+
+function setContainerMockState(newState: any) {
+  containerMockState = newState
+}
+
+const containerMock = {
+  pause: jest.fn(),
+  unpause: jest.fn(),
+  inspect: jest
+    .fn()
+    .mockImplementation(async () => ({ State: containerMockState })),
+}
+
+const dockerMock = {
+  getContainer: jest.fn().mockReturnValue(containerMock),
+}
+
+jest.mock('dockerode', () => jest.fn().mockReturnValue(dockerMock))
+jest.genMockFromModule('execa')
+jest.mock('execa')
+import Dockerode from 'dockerode'
+import execa from 'execa'
+import { assert } from 'chai'
+import ChainlinkClient from './chainlinkClient'
+;(execa.sync as jest.Mock).mockReturnValue({ stdout: '{}' })
+
+let clClient: ChainlinkClient
+
+const CLIENT_NAME = 'test_name'
+const CLIENT_ENDPOINT = 'http://test.com'
+const CONTAINER_NAME = 'test_container'
+const ENV = { env: { CLIENT_NODE_URL: 'http://test.com', ROOT: '~/test_name' } }
+
+beforeEach(() => {
+  clClient = new ChainlinkClient(CLIENT_NAME, CLIENT_ENDPOINT, CONTAINER_NAME)
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('ChainlinkClient', () => {
+  describe('#constructor', () => {
+    it('sets the correct properties on the instance', () => {
+      expect(Dockerode).toHaveBeenCalledTimes(1)
+      expect(dockerMock.getContainer).toHaveBeenCalledTimes(1)
+      expect(dockerMock.getContainer).toHaveBeenCalledWith(CONTAINER_NAME)
+      assert.equal(clClient.name, CLIENT_NAME)
+      assert.equal(clClient.clientURL, CLIENT_ENDPOINT)
+      assert.equal(clClient.clientURL, CLIENT_ENDPOINT)
+      // assert.equal(clClient.container, containerMock)
+      assert.equal(clClient.rootDir, `~/${CLIENT_NAME}`)
+    })
+  })
+
+  describe('#pause', () => {
+    it('calls the correct docker functions', async () => {
+      setContainerMockState({ Paused: false })
+      await clClient.pause()
+      expect(containerMock.inspect).toHaveBeenCalledTimes(1)
+      expect(containerMock.pause).toHaveBeenCalledTimes(1)
```

Comment:
Do we also want to assert that the value of `pause` is now `true`?

Reply:
I don't think that makes sense because their is no local state in the `ChainlinkClient` class; the state comes from the actual container. We'd have to mock the return value so all we could really test for here is that we built a mocked state value correctly.

Reply:
Makes sense, fair enough